projects/ADuCM3029_demo_cn0326/src/CN0326.c

```diff
@@ -0,0 +1,499 @@
+	i32voltage = AD7793_ConvertToVolts(ui32adcValue);
+
+	/* Calculate pH value */
+	ph = PH_ISO - (((double)((i32voltage - TOLERANCE) * FARADAY_CONST))
+			/ (PH_CONST * AVOGADRO_NUMBER * (temp + K_DEGREES)));
+
+	return ph;
+}
+
+
+
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+
+/**
+   @brief Display info for <help> command
+
+   @param args - pointer to the arguments on the command line.
+
+   @return none
+**/
+void CN0326_CmdHelp(uint8_t *args)
+{
+	printf("\n");
+	printf("Available commands:\n");
+	printf(" help                   - Display available commands\n");
+	printf(" calibrate <ch>         - Calibrate selected channel or all channels. \n");
+	printf("                          <ch> = AIN1, AIN2, AIN3 or all \n");
+	printf(" temp                   - Display temperature value\n");
+	printf(" ph                     - Display pH value\n");
+	printf(" reset                  - Reset ADC converter\n");
+}
+
+/**
+   @brief Display info for <temp> command
+
+   @param args - pointer to the arguments on the command line.
+
+   @return none
+**/
+void CN0326_CmdTemp(uint8_t *args)
+{
+	float temp;
+
+	temp = CN0326_CalculateTemp();
+
+	printf("Temperature = %.2f[˚C]\n", temp);
+}
+
+
+/**
+   @brief Display info for <ph> command
+
+   @param args - pointer to the arguments on the command line.
+
+   @return none
+**/
+void CN0326_CmdPH(uint8_t *args)
+{
+	float ph;
+
+	ph = CN0326_CalculatePH();
+
+	printf("pH = %.2f\n", ph);
+}
+
+
+/**
+   @brief Display info for <calibrate> command
+
+   @param args - pointer to the arguments on the command line.
+
+   @return none
+**/
+void CN0326_CmdCalibration(uint8_t *args)
+{
+
+	uint8_t  *p = args;
+	char     arg[5];
+	uint8_t ui8channel;
+
+	/* Check if this function gets an argument */
+	while (*(p = CN0326_FindArgv(p)) != '\0') {
+		/* Save channel parameter */
+		CN0326_GetArgv(arg, p);
+	}
+
+	if(strncmp(arg, "AIN1", 5) == 0) {
+		AD7793_Calibrate(AD7793_CH_AIN1P_AIN1M, CAL_INT_ZERO_MODE);
+		AD7793_Calibrate(AD7793_CH_AIN1P_AIN1M, CAL_INT_FULL_MODE);
+
+		timer_sleep(300);
+
+		printf("Calibration completed for %s channel!\n", arg);
+	} else if(strncmp(arg, "AIN2", 5) == 0) {
+		AD7793_Calibrate(AD7793_CH_AIN2P_AIN2M, CAL_INT_ZERO_MODE);
+		AD7793_Calibrate(AD7793_CH_AIN2P_AIN2M, CAL_INT_FULL_MODE);
+
+		timer_sleep(300);
+
+		printf("Calibration completed for %s channel!\n", arg);
+	} else if(strncmp(arg, "AIN3", 5) == 0) {
+		AD7793_Calibrate(AD7793_CH_AIN3P_AIN3M, CAL_INT_ZERO_MODE);
+		AD7793_Calibrate(AD7793_CH_AIN3P_AIN3M, CAL_INT_FULL_MODE);
+
+		timer_sleep(300);
+
+		printf("Calibration completed for %s channel!\n", arg);
+	} else if(strncmp(arg, "all", 4) == 0) {
+		for(ui8channel = 0; ui8channel <= 2; ui8channel++) {
+
+			AD7793_Calibrate(ui8channel, CAL_INT_ZERO_MODE);
+			AD7793_Calibrate(ui8channel, CAL_INT_FULL_MODE);
+		}
+
+		timer_sleep(300);
+
+		printf("Calibration completed for %s channels!\n", arg);
+	} else {
```

Comment:
I would use a` switch(case)` instead of several `else/else if` statements, as it makes the code more readable